### PR TITLE
Systemjs depcache output

### PR DESCRIPTION
**Pull request by mattquinlan440**
# Thanks for opening a Pull Request

## Changes

Please outline the reason for the changes you are making:

This PR adds the ability to output the scripts in a `depcache` format that's usable with single SPA / systemjs implementations

[Change description here]

Are there any breaking changes you are aware of in the PR?

Not directly. Updating arborist a major version, but its downstream change removed node versions we already didn't support.

## General Checklist

* [x] Change is tested locally
* [x] Demo is updated to exercise change (if applicable)
* [x] [WIP] flag is removed from the title

[1]:https://docs.npmjs.com/about-semantic-versioning
[2]:https://docs.npmjs.com/cli/deprecate
[3]:https://github.com/meltwater/applet-orchard/blob/master/docs/README.md

### CODE REVIEW

File: src/build-depcache-output/resolve-required-dependency-scripts/index.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    Logger.info(`Creating scripts for ${externalPackageEntry.packageName}`);
+
+    return externalPackageEntry.getEsmUrls(version);
+}
+
+export function resolveRequiredDependencyScripts({ dependencies, dependencyMap }) {
+    ac.assertType(dependencies, Object, 'dependencies');
+    ac.assertType(dependencyMap, Object, 'dependencyMap');
+
+    const scriptsArray = dependencies.reduce((scriptsArray, dependency) => {
```
Comment (rajeshsusai) [inline src/build-depcache-output/resolve-required-dependency-scripts/index.js]: Nitpick: I  like to distinguish the name of my accumulator for readability purposes, so seeing two `scriptsArray` kinda throws me off haha

Comment (mattquinlan440) [reply src/build-depcache-output/resolve-required-dependency-scripts/index.js]: Good call! I agree, that will definitely help readability. I'll make a change

File: src/cli/index.js [hunk 2]
```diff
@@ -46,11 +47,20 @@ sywac
...
+        desc: 'Whether to output for use with systemjs depcache or not',
+        defaultValue: false
+    })
     .help('-h, --help')
     .parseAndExit()
     .then(options => ({ ...options, logging: loggingMap[options.logging] }))
     .then(options => new CliOptions(options))
-    .then(cliOptions => buildAppOutput(cliOptions))
+    .then(cliOptions => {
+        if(cliOptions.outputDepcache) {
```
Comment (jtheriault) [inline src/cli/index.js:59]: Could we get a description and/or sample in the README of what this is and does?

Comment (mattquinlan440) [reply src/cli/index.js:59]: Added something to the README. Let me know if it's enough, or more details are needed

**Review by krishna-koushik — APPROVED** [review]

File: src/cli/index.js (hunk 2 above)
Comment (jtheriault) [reply src/cli/index.js:59]: That does the job nicely. Thank you

**Review by jtheriault — APPROVED** [review]
This is a good extension and also a good application of "1, 2, n." I think there's some opportunity identifying some common abstractions when we add an "nth" output type, for example import maps.

**Pull request merged**